include/mesh/mesh.h

```diff
@@ -79,10 +79,16 @@ class Mesh : public DefaultMesh
   Mesh(Mesh &&) = delete;
 
   /**
-   * Copy and move assignment are not allowed.
+   * Copy assignment is not allowed.
    */
   Mesh & operator= (const Mesh &) = delete;
 
+  /**
+   * Move assignment is allowed but mesh objects
+   * should be moved using the MeshBase::assign function.
```

Comment:
I wouldn't say "should be" here.  This assignment operator ought to work perfectly.  We can't use it because we only have internal access to a MeshBase, but for anyone writing app code with access to the subclass there'd be no reason to prefer the virtual move over the standard move.